src/mbgl/text/collision_tile.cpp

```diff
@@ -20,7 +20,11 @@ CollisionTile::CollisionTile(PlacementConfig config_) : config(std::move(config_
     rotationMatrix = { { angle_cos, -angle_sin, angle_sin, angle_cos } };
     reverseRotationMatrix = { { angle_cos, angle_sin, -angle_sin, angle_cos } };
 
-    perspectiveRatio = 1.0f + 0.5f * ((config.cameraToTileDistance / config.cameraToCenterDistance) - 1.0f);
+    perspectiveRatio =
+        1.0f +
+        0.5f * (util::division(config.cameraToTileDistance, config.cameraToCenterDistance, 0.0f) -
```

Comment:
In the event `cameraToCenterDistance` were set to 0 (which shouldn't happen), this would return a `perspectiveRatio` of `0.5`. It probably makes slightly more sense to make the `nan` value `1.0f` so the perspective ratio comes out to a neutral `1.0`.